Guard against high `ulimit -n` in when starting vnc

Earlopain opened the pull request:
### Description
See #2045. Basically what https://github.com/SeleniumHQ/docker-selenium/issues/2045#issuecomment-1849497065 proposed. If the ulimit is high, defensively lower it.

### Motivation and Context
Recent versions of docker in combination with the upstream systemd unit files pass an incredibly high `ulimit -n` to the docker container, up to 1 billion. That causes minute high delays and CPU spinning when connecting to VNC while it enumerates all the file descriptors.

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [contributing](https://selenium.dev/documentation/en/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
<!--- Provide a general summary of your changes in the Title above -->

diemol commented on NodeBase/start-vnc.sh:
```diff
@@ -44,6 +44,17 @@ if [ "${START_XVFB:-$SE_START_XVFB}" = true ] ; then
       echo "Waiting for Xvfb..."
     done
 
+    # Guard against unreasonably high nofile limits. See https://github.com/SeleniumHQ/docker-selenium/issues/2045
+    if [[ $(ulimit -n) -gt 200000 ]]; then
+      echo "Trying to lower open file descriptor limit from $(ulimit -n) to 65536."
+      ulimit -n 65536
+      if [ $? -eq 0 ]; then
+        echo "Successfully lowered the open descriptor limit."
+      else
+        echo "Failed to lower the open file descriptor limit. This can result in delays when connecting to VNC."
+      fi
+    fi
+
```
What happens if the command fails? Does execution break or it just resumes and echoes the 2nd message?

Earlopain replied:
It just carries on, this file doesn't have stop on error enabled. A few lines above is a similar construct.

I'm not even 100% sure that this command can even fail. Lowering the limit seems like something that always works but I'm not knowledgeable enough to say for certain so I added a bit of logging.

VietND96 replied:
I have tried a negative test by update wrong command, 2nd message echoes and next step can go
```bash
Trying to lower open file descriptor limit from 1048576 to 65536.
/opt/bin/start-vnc.sh: line 50: ulimit: -1: invalid option
ulimit: usage: ulimit [-SHabcdefiklmnpqrstuvxPT] [limit]
Failed to lower the open file descriptor limit. This can result in delays when connecting to VNC.
11/12/2023 16:56:48 passing arg to libvncserver: -rfbport
11/12/2023 16:56:48 passing arg to libvncserver: 5900
11/12/2023 16:56:48 passing arg to libvncserver: -rfbportv6
11/12/2023 16:56:48 passing arg to libvncserver: 5900
11/12/2023 16:56:48 -usepw: found /home/seluser/.vnc/passwd

```

diemol replied:
If we are not sure if the command can fail, wouldn't it be a misleading message?

Earlopain replied:
The file descriptor limit has been found to be high and lowering it failed, for whatever reason. That's exactly what the message is warning you about, if say your limit is set to 1 billion for one reason or another. Continuing on the user should be aware that things might not work like they expect them to.

The output above from vietnd96 introduced an intentional invalid command and it'll never look like that with the actual code.

VietND96 replied:
My test only verify status != 0 not break the code. To avoid misleading, I think we can change another message without word `Failed` something like
`echo "The open file descriptor limit could not be updated. This can result in delays when connecting to VNC."`

VietND96 replied:
I think the limit value, we can get from ENV or default, it helps user is able to set another value e.g `-e SE_VNC_ULIMIT=100000`
something like `ulimit -n ${SE_VNC_ULIMIT:-65536}`

Earlopain replied:
Sure, done. Your version sounds better anyways

Earlopain replied:
> I think the limit value, we can get from ENV or default, it helps user is able to set another value e.g `-e SE_VNC_ULIMIT=100000` something like `ulimit -n ${SE_VNC_ULIMIT:-65536}`

I pushed a commit, check it out. I dropped the notice about VNC connection delays since that isn't always applicable anymore.

Earlopain committed:
Guard against high `ulimit -n` when starting vnc

Recent versions of docker in combination with the upstream systemd
unit files pass an incredibly high `ulimit -n` to the docker container, up
to 1 billion. That causes minute high delays and CPU spinning when
connecting to VNC while it enumerates all the file descriptors.

See #2045

VietND96 committed:
Merge branch 'trunk' into vnc-ulimit

Earlopain committed:
Update failure message

Earlopain committed:
Allow the ulimit to be configurable by env

VietND96 commented:
Thank you, @Earlopain!
